flash/vision/detection/model.py

```diff
@@ -81,7 +94,51 @@ def training_step(self, batch, batch_idx) -> Any:
         # fasterrcnn takes both images and targets for training, returns loss_dict
         loss_dict = self.model(images, targets)
         loss = sum(loss_dict.values())
-        for k, v in loss_dict.items():
-            self.log("train_k", v)
-
+        self.log_dict({f"train_{k}": v for k, v in loss_dict.items()}, on_step=True, on_epoch=True, prog_bar=True)
         return loss
+
+    def validation_step(self, batch, batch_idx):
```

Comment:
I don't think you have to override those functions.

Reply:
We would need to override the validation_step and test_step methods, as we don’t have to provide loss fn to faster rcnn. And it would throw an error if we use the default step method of Task class.